src/Profiles.ts

```diff
@@ -300,59 +278,27 @@ export class Profiles { // Processing stops if there are no profiles detected
                 loadSession.password = passWord.trim();
             }
         }
-
         const updSession = await zowe.ZosmfSession.createBasicZosmfSession(loadSession as IProfile);
         return [updSession.ISession.user, updSession.ISession.password, updSession.ISession.base64EncodedAuth];
     }
 
     private async saveProfile(ProfileInfo, ProfileName, ProfileType) {
-        const mainZoweDir = path.join(require.resolve("@brightside/core"), "..", "..", "..", "..");
-        // we have to mock a few things to get the Imperative.init to work properly
-        try {
-            (process.mainModule as any).filename = require.resolve("@brightside/core");
-        } catch (error) {
-            vscode.window.showErrorMessage(error.message);
-        }
-        try {
-            ((process.mainModule as any).paths as any).unshift(mainZoweDir);
-        } catch (error) {
-            vscode.window.showErrorMessage(error.message);
-        }
-        if (this.initValue === -1) {
-            try {
-            // we need to call Imperative.init so that any installed credential manager plugins are loaded
-                await Imperative.init({ configurationModule: require.resolve("@brightside/core/lib/imperative.js") });
-            } catch (error) {
-                vscode.window.showErrorMessage(error.message);
-            }
-            this.initValue = 0;
-        }
         let zosmfProfile: IProfile;
         try {
-            zosmfProfile = await new CliProfileManager({
-                profileRootDirectory: path.join(ImperativeConfig.instance.cliHome, "profiles"),
-                type: "zosmf"
-            }).save({ profile: ProfileInfo, name: ProfileName, type: ProfileType });
+            zosmfProfile = await (await this.getCliProfileManager(ProfileType)).save({ profile: ProfileInfo, name: ProfileName, type: ProfileType });
         } catch (error) {
             vscode.window.showErrorMessage(error.message);
         }
         return zosmfProfile.profile;
     }
 
-    private isSpawnReqd() {
-        if (this.spawnValue === -1) {
-            const homedir = os.homedir();
-            this.spawnValue = 0;
-            try {
-                const fileName = path.join(homedir, ".zowe", "settings", "imperative.json");
-                const settings = JSON.parse(fs.readFileSync(fileName).toString());
-                const value = settings.overrides.CredentialManager;
-                this.spawnValue = value !== false ? 0 : 1;
-            } catch (error) {
-                // default to spawn
-                this.spawnValue = 0;
-            }
+    private async getCliProfileManager( type: string ) {
+        if (!this.profileManager) {
+            this.profileManager = await new CliProfileManager({
+                profileRootDirectory: path.join(os.homedir(), ".zowe", "profiles"),
```

Comment:
It's not always safe to use `os.homedir()` here. If the environment variable `ZOWE_CLI_HOME` is set, it should take precedence.

Previously `ImperativeConfig.instance.cliHome` was used to handle this, but it is no longer convenient to use because it requires that `Imperative.init` has already been called.

The following hack would work by mocking out the `loadedConfig` object, but hopefully someone has a better idea :)
```
ImperativeConfig.instance.loadedConfig = {
    defaultHome: path.join(os.homedir(), ".zowe"),
    envVariablePrefix: "ZOWE"
};
const zoweDir = ImperativeConfig.instance.cliHome;
```

Reply:
Thanks Timothy I can add that into the code. My tests so far show that Imperative had not been initialized when the Config is referenced. Is there a way that I can do that prior to calling these functions?

Reply:
Discussed this with @zFernand0, it's good that the code initializing Imperative was removed in this PR because it improves performance :) This is really a defect in Imperative, it shouldn't be necessary to call its initialize function just to access constants like the CLI home directory.

Reply:
Created an issue in Imperative to track this: https://github.com/zowe/imperative/issues/357